src/controller/java/AndroidCallbacks.cpp

```diff
@@ -156,15 +106,15 @@ void GetConnectedDeviceCallback::OnDeviceConnectedFn(void * context, Messaging::
     auto * self          = static_cast<GetConnectedDeviceCallback *>(context);
     jobject javaCallback = self->mJavaCallbackRef;
     JniLocalReferenceManager manager(env);
-    // Release global ref so application can clean up.
-    env->DeleteGlobalRef(self->mWrapperCallbackRef);
+    // Release wrapper's global ref so application can clean up the actual callback underneath.
+    VerifyOrReturn(self->mWrapperCallbackRef != nullptr, ChipLogError(Controller, "mWrapperCallbackRef is null"));
+    JniObject wrapper(env, self->mWrapperCallbackRef);
```

Comment:
suggest: 

make the constructor for the wrapper object take a unique_ptr or use some other mechanism to indicate passing of ownership

Reply:
Done, thanks